### PR TITLE
Don't unpack from the bundle CLI when multiple archives are uploaded

### PR DESCRIPTION
**Pull request by epicfaace**
### Reasons for making this change

For https://github.com/codalab/codalab-worksheets/issues/3439. Don't unpack from the bundle CLI when multiple archives are uploaded.

**This is a breaking change.** This means that running, for example, `cl upload item.gz dir.tar.gz` will not unpack the each individual archive file; rather, these files will just be added as usual to the entire bundle directory.

I've also added unit tests that exercise all of the functionality of `zip_util.pack_file_for_upload`, as we'll be changing this function later for streaming (https://github.com/codalab/codalab-worksheets/issues/3437).

### CODE REVIEW

**Review by adiprerepa — COMMENTED** [review]
why are we doing this?

**Comment by epicfaace** [comment]
See https://docs.google.com/document/d/1l4kOqi9irBjOApmn4E6vlzsjAXDJbetIyVw8gMRHrpU/edit# for the design doc. Basically, this is not a feature that is used very often; it's not really a common use case for users of the CodaLab CLI.

It makes it harder to implement Blob Storage and/or streaming uploads, because in order to create streaming uploads with multiple files (i.e. `cl upload a.txt b.txt c.txt`), we have to construct an archive on-the-fly with `a.txt`, `b.txt`, and `c.txt` in it.

On the other hand, if we kept the original behavior of also unpacking each archive source, a command such as `cl upload a.tar.gz b.tar.gz c.tar.gz` would require us to, while we write to the output .tar.gz file in a stream, 1) unpack `a.tar.gz`, then for every item in `a.tar.gz`, prepend `a/` to its path, and add it to the final output archive in the stream; 2) do the same with `b.tar.gz`, but prepend `b/` to these paths; and 3) do the same with `c.tar.gz`. A lot of complexity, and it's possible, but it's probably not worth the effort.

File: tests/unit/lib/zip_util_test.py [hunk 1]
```diff
@@ -1,37 +1,211 @@
...
         self.assertEqual(get_archive_ext(path), ".tar.gz")
         self.assertEqual(strip_archive_ext(path), "https://codalab.org/file")
+
+    def test_pack_single_file(self):
+        """Pack a single file."""
+        with tempfile.NamedTemporaryFile() as f:
+            f.write(SAMPLE_CONTENTS)
+            f.flush()
+            packed = pack_files_for_upload(
+                sources=[f.name], should_unpack=False, follow_symlinks=False
```
Comment (jzwang43) [inline tests/unit/lib/zip_util_test.py:69]: Do we still need `should_unpack` since we are not doing unpack anymore?

Comment (epicfaace) [reply tests/unit/lib/zip_util_test.py:69]: We still unpack single archives (such as `cl upload a.zip`). We're just not unpacking archives if multiple ones are specified.

**Review by jzwang43 — APPROVED** [review]
Looks good.

**Pull request merged**